troubadix/plugins/illegal_characters.py

```diff
@@ -30,27 +36,60 @@
 # | =               are delimiter in the internal VT cache
 #                   (e.g. Tag1=Foo|Tag2=Bar|Tag3=Baz)
 #                   in all script_tag(name:"", value:"")
-FORBIDDEN_CHARS = ["|", "=", ";"]
+FORBIDDEN_CHARS = ["|", ";"]
+WARNING_CHARS = ["="]
 
 # replacement character will be a whitespace
 REPLACE_CHAR = " "
```

Comment:
Not sure if this would be in the scope of this PR (that's my only as a general comment and not a change request):

Replacing any of these chars with just a space when passing `--fix` to Troubadix should be avoided because it changes the meaning of the metadata (e.g. a `|` is often used in code examples of LSCs).

Would it be possible "while on it" to have some kind of array in `FORBIDDEN_CHARS` and `WARNING_CHARS` which includes the char which is an error or warning and the char which should be used for a replacement (e.g. replacing `|` with `<pipe>`?

Reply:
Sounds good to me, so sth. like
```
FORBIDDEN_CHARS = {
    "|": "<pipe>",
    ";": "foobar"
}
```
Then the check for existence of a character could be kept as is and only replacing it would need to be adjusted (accessing the corresponding dict index instead)

Reply:
In conjunction with Nicolas request from above, regarding a single check:
We could create a dictionary like this: 
```
forbidden_chars = {
    <char>: (<ErrorType>, <Replacement>)
}
```

Which would wrap warning and error in the same list and allow a specific replace per forbidden char.

Reply:
@cfi-gb Any desired replacement for `;` ?

Reply:
Side note: In Notus Generator we replace `;` with `,`

Reply:
Replacing `;` with `,` by default should work. The dev would need to review the changes anyway and could still change it if it should be replaced by e.g. `<semicolon>`.

Reply:
`,` it is then (for now)